src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the MLflow model registry for ZenML."""
+
+
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+import mlflow
+from mlflow import MlflowClient
+from mlflow.exceptions import MlflowException
+from mlflow.pyfunc import load_model
+
+from zenml import __version__
+from zenml.client import Client
+from zenml.enums import StackComponentType
+from zenml.integrations.mlflow.experiment_trackers.mlflow_experiment_tracker import (
+    MLFlowExperimentTracker,
+)
+from zenml.integrations.mlflow.flavors.mlflow_model_registry_flavor import (
+    MLFlowModelRegistryConfig,
+)
+from zenml.integrations.mlflow.mlflow_utils import (
+    get_missing_mlflow_experiment_tracker_error,
+)
+from zenml.logger import get_logger
+from zenml.model_registries.base_model_registry import (
+    BaseModelRegistry,
+    ModelRegistration,
+    ModelVersion,
+    ModelVersionStage,
+)
+from zenml.stack.stack import Stack
+from zenml.stack.stack_validator import StackValidator
+
+logger = get_logger(__name__)
+
+
+class MLFlowModelRegistry(BaseModelRegistry):
+    """Register models using MLflow."""
+
+    _client: Optional[MlflowClient] = None
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the model registry and validate the tracking uri.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(*args, **kwargs)
+
+    @property
+    def config(self) -> MLFlowModelRegistryConfig:
+        """Returns the `MLFlowModelRegistryConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(MLFlowModelRegistryConfig, self._config)
+
+    def configure_mlflow(self) -> None:
+        """Configures the MLflow Client with the experiment tracker config."""
+        experiment_tracker = Client().active_stack.experiment_tracker
+        if not isinstance(experiment_tracker, MLFlowExperimentTracker):
```

Comment:
Can also assert this without raising the exception I guess, this will always by covered by the stack validator